source/modules/sensors/SensorsGetters.mc

```diff
@@ -268,16 +268,24 @@ module SensorsGetters {
             return System.getDeviceSettings().phoneConnected;
         }
 
-        function getHR() as Number? {
+        function getHR() as Array<Number?>? {
             var activityInfo = Activity.getActivityInfo();
             var value = activityInfo != null ? activityInfo.currentHeartRate : null;
-
+            var isHeartWorking = true;
```

Comment:
Don't forget to clean up after these variables ;)

Reply:
Oh my god！It may be 2 a.m. where you are now, and you haven't slept yet.

Reply:
it's 3 am :) a bit sleepy was thinking about the problem of updating data from OW and already made a fix, so I'm going to sleep =D

Reply:
Sorry, there may be some errors in our understanding of how heart rate works.
If the heart rate is obtained from SensorHistory.getHeartRateHistory({}).next(), I think the heart rate is not working.
It is working only when the heart rate is obtained in Activity.getActivityInfo().currentHeartRate (1), otherwise it is not working (0).
Therefore, the isHeartWorking variable still needs to be retained.